pkg/plugins/manifest.go

```diff
@@ -111,23 +167,50 @@ func getPluginSignatureState(log log.Logger, plugin *PluginBase) PluginSignature
 		fp := filepath.Join(plugin.PluginDir, p)
 		f, err := os.Open(fp)
 		if err != nil {
-			return PluginSignatureModified
+			log.Warn("Plugin file listed in the manifest was not found", "plugin", plugin.Id, "filename", p, "dir", plugin.PluginDir)
+			return &PluginSignatureState{
+				Status: pluginSignatureModified,
+			}, nil
 		}
 		defer f.Close()
 
 		h := sha256.New()
 		if _, err := io.Copy(h, f); err != nil {
 			log.Warn("Couldn't read plugin file", "plugin", plugin.Id, "filename", fp)
-			return PluginSignatureModified
+			return &PluginSignatureState{
+				Status: pluginSignatureModified,
+			}, nil
 		}
 		sum := hex.EncodeToString(h.Sum(nil))
 		if sum != hash {
 			log.Warn("Plugin file's signature has been modified versus manifest", "plugin", plugin.Id, "filename", fp)
-			return PluginSignatureModified
+			return &PluginSignatureState{
+				Status: pluginSignatureModified,
+			}, nil
 		}
+		pluginFileInSignature[p] = true
+	}
+
+	// Track files that were found but not listed in the manifest
+	var unsignedFiles []string
+	for f := range pluginFileInSignature {
+		if inSignature := pluginFileInSignature[f]; !inSignature && manifest.isV2() {
+			unsignedFiles = append(unsignedFiles, f)
+		}
+	}
+
+	if len(unsignedFiles) > 0 {
+		log.Warn("The following files were not included in the signature", "plugin", plugin.Id, "files", unsignedFiles)
+		return &PluginSignatureState{
+			Status: pluginSignatureModified,
+		}, nil
```

Comment:
There shouldn't be any need to execute any of this code if it isn't v2.
```suggestion
	if manifest.isV2() {
		var unsignedFiles []string
		for f := range pluginFileInSignature {
			if !pluginFileInSignature[f] {
				unsignedFiles = append(unsignedFiles, f)
			}
		}
		
		if len(unsignedFiles) > 0 {
			log.Warn("The following files were not included in the signature", "plugin", plugin.Id, "files", unsignedFiles)
			return &PluginSignatureState{
				Status: pluginSignatureModified,
			}, nil
```